pkg/thing/interactors/register_thing.go

```diff
@@ -82,25 +82,27 @@ func (i *ThingInteractor) Register(authorization, id, name string) error {
 	if err != nil {
 		errReply := i.reply(id, "", err)
 		if errReply != nil {
-			i.logger.Error(errReply)
+			i.logger.Debug(err)
+			return errReply
 		}
-
 		return err
 	}
 
 	// Get the id generated as a token and send in the response
 	token, err := i.thingProxy.Create(id, name, authorization)
-	errReply := i.reply(id, token, err)
-	if errReply != nil {
-		i.logger.Error(errReply)
-		return errReply
+	if err != nil {
+		errReply := i.reply(id, token, err)
```

Comment:
You can remove this call since it is already outside the if statement

Reply:
I didn't quite catch why, could you explain? :eyes: 

From my understanding, if there is only one `i.reply(id, token, errorVariable)`, `Register()` wouldn't be able to notify the client about either: 

1. the registration error in case `i.thingProxy.Create(id, name, authorization)` 
2. the error in case `i.connectorPublisher.SendRegisterDevice(id, name)`

Reply:
I didn't quite catch why, could you explain? :eyes: 

From my understanding, if there is only one `i.reply(id, token, errorVariable)`, `Register()` wouldn't be able to notify the client about either: 

1. the registration error in case `i.thingProxy.Create(id, name, authorization)` 
2. the error in case `i.connectorPublisher.SendRegisterDevice(id, name)`

Reply:
He was talking about the same method being called [here](https://github.com/CESARBR/knot-babeltower/pull/46/commits/de95a6815fa490dda27198ce1028eb0eeeb8a8cc#diff-60df0fe9cfe391c1945e90caf147ab91R102).

Reply:
@ramonhpr You are right! I replace the method call to right after the `Create` method call. Could you check if it's ok? :smile:

Reply:
@esdrasjnr did you push this change?

Reply:
@joaoaneto Yes! Note that I only replace the `reply` function call. This was necessary to maintain the desired flow. However, the changes you suggested will change it soon :smiley: